### PR TITLE
fix: capture PR number for Azure

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [CYCLOUD-765](https://cypress-io.atlassian.net/browse/CYCLOUD-765)

### Additional details

Capture `SYSTEM_PULLREQUEST_PULLREQUESTNUMBER` to display the value in the cloud.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

[CYCLOUD-765]: https://cypress-io.atlassian.net/browse/CYCLOUD-765?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ _Released 03/28/2023 (PENDING)_
 **Misc:**
 
  - Made some minor styling updates to the Debug page. Addresses [#26041](https://github.com/cypress-io/cypress/issues/26041).
+ - Capture the [Buildkite](https://buildkite.com/) CI provider's environment variable `BUILDKITE_RETRY_COUNT` to handle CI retries in the Cloud. Addressed in [25750](https://github.com/cypress-io/cypress/pull/25750).
+ - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the value in the Cloud. Addressed in [26215](https://github.com/cypress-io/cypress/pull/26215)
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]: I expect this to be either a bug fix or feature enhancement, not listed as a misc update

Comment (estrada9166) [reply cli/CHANGELOG.md]: I just updated it to be under Bugfixes

Comment (estrada9166) [reply cli/CHANGELOG.md]: I just updated it to be under Bugfixes

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,11 +7,14 @@ _Released 03/28/2023 (PENDING)_
 
  - Fixed a compatibility issue so that component test projects can use [Vite](https://vitejs.dev/) version 4.2.0 and greater. Fixes [#26138](https://github.com/cypress-io/cypress/issues/26138).
  - Changed the way that Git hashes are loaded so that non-relevant runs are excluded from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).
+ - Capture the [Buildkite](https://buildkite.com/) CI provider's environment variable `BUILDKITE_RETRY_COUNT` to handle CI retries in the Cloud. Addressed in [25750](https://github.com/cypress-io/cypress/pull/25750).
+ - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the value in the Cloud. Addressed in [26215](https://github.com/cypress-io/cypress/pull/26215).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
 - Capture the [Buildkite](https://buildkite.com/) CI provider's environment variable `BUILDKITE_RETRY_COUNT` to handle CI retries in the Cloud. Addressed in [#25750](https://github.com/cypress-io/cypress/pull/25750).
 - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the value in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
```

**Comment by emilyrohrbough** [comment]
@estrada9166 Are there doc changes that need to go with this change? or do we already document this ENV?

**Comment by emilyrohrbough** [comment]
@estrada9166 Are there doc changes that need to go with this change? or do we already document this ENV?

**Comment by estrada9166** [comment]
> @estrada9166 Are there doc changes that need to go with this change? or do we already document this ENV?

@emilyrohrbough these env variables are not documented - we only document the ones related with the CI-BUILD-ID

**Comment by emilyrohrbough** [comment]
@estrada9166 why wouldn't we document all envs we are capturing?

**Review by lmiller1990 — APPROVED** [review]
Code seems fine!

**Comment by lmiller1990** [comment]
> @estrada9166 why wouldn't we document all envs we are capturing?

I'm guessing we don't document things considered internal to CI providers - users should be able to move from CI to CI seamlessly, I think these env vars would be considered an implementation detail. Is there any value in adding something like this to the documentation?

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ _Released 03/28/2023_
  - Changed the way that Git hashes are loaded so that non-relevant runs are excluded from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).
  - Corrected the [`.type()`](https://docs.cypress.io/api/commands/type) command to account for shadow root elements when determining whether or not focus needs to be simulated before typing. Fixes [#26198](https://github.com/cypress-io/cypress/issues/26198).
  - Fixed an issue where an incorrect working directory could be used for Git operations on Windows. Fixes [#23317](https://github.com/cypress-io/cypress/issues/23317).
+ - Capture the [Buildkite](https://buildkite.com/) CI provider's environment variable `BUILDKITE_RETRY_COUNT` to handle CI retries in the Cloud. Addressed in [#25750](https://github.com/cypress-io/cypress/pull/25750).
+ - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the value in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
 - Capture the [Azure](https://azure.microsoft.com/) CI provider's environment variable [`SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) to display the linked PR number in the Cloud. Addressed in [#26215](https://github.com/cypress-io/cypress/pull/26215).
```

Comment (emilyrohrbough) [reply cli/CHANGELOG.md]: This entry needs moved to the unreleased section

**Comment by elylucas** [comment]
> @estrada9166 why wouldn't we document all envs we are capturing?

as these are values captured during CI and sent to Cloud for our own purposes, I don't think they need to be publicly documented

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**